app/src/main/java/com/xinto/opencord/ui/widget/Message.kt

```diff
@@ -36,6 +36,7 @@ fun WidgetChatMessage(
     content: (@Composable () -> Unit)? = null,
     attachments: (@Composable () -> Unit)? = null,
     embeds: (@Composable () -> Unit)? = null,
+    isMerged: Boolean = false
```

Comment:
It is way more logical for this variable to not exist and instead rely on author and avatar nullability, message component should have no idea about what "merged" messages are

Reply:
guhhh i blame rushie for making me change it

Reply:
skill issue

Reply:
guh wrong
how is it logical for layout changing based on if 2 certain inputs are null

Reply:
plus, isn't it more rerendering if they change

Reply:
> guh wrong how is it logical for layout changing based on if 2 certain inputs are null

Because the layout is supposed to be adaptive, as well as it should not expose anything that directly affects the constraints of the layout. Also, the widget (which will be renamed to component in the future because widget has a very different meaning which doesn't suit the customizability of the WidgetMessage) should not know anything about discord and/or merged messages, the only thing it should know is how to constrain the layout based on the inputs it already receives, I think this is the best way to encapsulate data as well as make the widget (component) as reusable as possible

Reply:
> plus, isn't it more rerendering if they change but the result stays the same

I don't understand this

Reply:
im going with xinto on this

Reply:
EXPLOOOOOD